essentials-framework/Essentials Devices Common/Essentials Devices Common/VideoCodec/ZoomRoom/ZoomRoom.cs

```diff
@@ -2818,16 +2820,25 @@ public void StartMeeting(uint duration)
 
         public void LeaveMeeting()
         {
-            SendText("zCommand Call Leave");
+			if (_meetingPasswordRequired) _meetingPasswordRequired = false;
+			if (_waitingForUserToAcceptOrRejectIncomingCall) _waitingForUserToAcceptOrRejectIncomingCall = false;
+			
+			SendText("zCommand Call Leave");
         }
 
 		public override void EndCall(CodecActiveCallItem call)
 		{
+			if (_meetingPasswordRequired) _meetingPasswordRequired = false;

```

Comment:
We already are monitoring the call state change and setting this value false.  We probably should not be modifying this when calling this method based on user input.  Wherever possible we should respond to device state changes.